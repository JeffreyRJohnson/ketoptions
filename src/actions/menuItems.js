import uuid from 'uuid';
import database from '../firebase/firebase';

// ADD_MENU_ITEM
export const addMenuItem = menuItem => ({
  type: 'ADD_MENU_ITEM',
  menuItem
});

export const startAddMenuItem = (menuItemData = {}) => {
  return dispatch => {
    const {
      calories = 0,
      carbs = 0,
      fat = 0,
      menu_item = '',
      protein = 0,
      restaurant_name = ''
    } = menuItemData;

    const menuItem = {
      calories,
      carbs,
      fat,
      menu_item,
      protein,
      restaurant_name
    };

    return database
      .ref('menuItems')
      .push(menuItem)
      .then(ref => {
        dispatch(
          addMenuItem({
            id: ref.key,
            ...menuItem
          })
        );
      });
  };
};

// REMOVE_MENU_ITEM
export const removeMenuItem = ({ id } = {}) => ({
  type: 'REMOVE_MENU_ITEM',
  id
});

// EDIT_MENU_ITEM
export const editMenuItem = (id, updates) => ({
  type: 'EDIT_MENU_ITEM',
  id,
  updates
});
