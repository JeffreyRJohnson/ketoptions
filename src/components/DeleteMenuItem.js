import React from 'react';

const DeleteMenuItem = () => <div>This is my delete Menu item component</div>;

export default DeleteMenuItem;
