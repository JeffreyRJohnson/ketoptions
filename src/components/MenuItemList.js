import React from 'react';
import { connect } from 'react-redux';
import MenuItemListItem from './MenuItemListItem';
import selectMenuItems from '../selectors/menuItems';

export const MenuItemList = props => (
  <div>
    {props.menuItems.length === 0 ? (
      <p>No menu items</p>
    ) : (
      props.menuItems.map(menuItem => {
        return <MenuItemListItem key={menuItem.id} {...menuItem} />;
      })
    )}
  </div>
);

const mapStateToProps = state => {
  return {
    menuItems: selectMenuItems(state.menuItems, state.filters)
  };
};

export default connect(mapStateToProps)(MenuItemList);
