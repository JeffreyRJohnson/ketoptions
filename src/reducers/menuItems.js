// MenuItems Reducer

const menuItemsReducerDefaultState = [];

export default (state = menuItemsReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_MENU_ITEM':
      return [...state, action.menuItem];
    case 'REMOVE_MENU_ITEM':
      return state.filter(({ id }) => id !== action.id);
    case 'EDIT_MENU_ITEM':
      return state.map(menuItem => {
        if (menuItem.id === action.id) {
          return {
            ...menuItem,
            ...action.updates
          };
        } else {
          return menuItem;
        }
      });
    default:
      return state;
  }
};
