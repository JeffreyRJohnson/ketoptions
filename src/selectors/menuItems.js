// Get visible menuItems

export default (menuItems, { text }) => {
  return menuItems.filter(menuItem => {
    const textMatch = menuItem.restaurant_name
      .toLowerCase()
      .includes(text.toLowerCase());

    return textMatch;
  });
};
