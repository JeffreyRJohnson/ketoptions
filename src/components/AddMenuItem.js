import React from 'react';
import { connect } from 'react-redux';
import MenuItemForm from './MenuItemForm';
import { startAddMenuItem } from '../actions/menuItems';

export class AddMenuItem extends React.Component {
  onSubmit = menuItem => {
    this.props.startAddMenuItem(menuItem);
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <h1>Add Menu Item</h1>
        <MenuItemForm onSubmit={this.onSubmit} />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startAddMenuItem: menuItem => dispatch(startAddMenuItem(menuItem))
});

export default connect(
  undefined,
  mapDispatchToProps
)(AddMenuItem);
