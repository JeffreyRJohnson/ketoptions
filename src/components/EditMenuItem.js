import React from 'react';
import { connect } from 'react-redux';
import MenuItemForm from './MenuItemForm';
import { editMenuItem, removeMenuItem } from '../actions/menuItems';

export class EditMenuItem extends React.Component {
  onSubmit = menuItem => {
    this.props.editMenuItem(this.props.menuItem.id, menuItem);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.removeMenuItem({ id: this.props.menuItem.id });
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <MenuItemForm menuItem={this.props.menuItem} onSubmit={this.onSubmit} />
        <button onClick={this.onRemove}>Remove</button>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  menuItem: state.menuItems.find(
    menuItem => menuItem.id === props.match.params.id
  )
});

const mapDispatchToProps = dispatch => ({
  editExpense: (id, menuItem) => dispatch(editMenuItem(id, menuItem)),
  removeMenuItem: data => dispatch(removeMenuItem(data))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditMenuItem);
