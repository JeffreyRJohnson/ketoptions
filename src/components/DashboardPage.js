import React from 'react';
import MenuItemList from './MenuItemList';
import MenuItemListFilters from './MenuItemListFilters';

const DashboardPage = () => (
  <div>
    <MenuItemListFilters />
    <MenuItemList />
  </div>
);

export default DashboardPage;
