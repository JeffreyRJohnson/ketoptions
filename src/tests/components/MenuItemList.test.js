import React from 'react';
import { shallow } from 'enzyme';
import { MenuItemList } from '../../components/MenuItemList';
import menuItems from '../fixtures/menuItems';

test('should render MenuItemList with menuItems', () => {
  const wrapper = shallow(<MenuItemList menuItems={menuItems} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render MenuItemList with empty message', () => {
  const wrapper = shallow(<MenuItemList menuItems={[]} />);
  expect(wrapper).toMatchSnapshot();
});
