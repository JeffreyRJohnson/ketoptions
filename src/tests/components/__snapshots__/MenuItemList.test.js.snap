// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MenuItemList with empty message 1`] = `
<div>
  <p>
    No menu items
  </p>
</div>
`;

exports[`should render MenuItemList with menuItems 1`] = `
<div>
  <MenuItemListItem
    calories={400}
    carbs={5}
    fat={20}
    id="1"
    key="1"
    menu_item="chilli - medium"
    protein={30}
    restaurant_name="Wendys"
  />
  <MenuItemListItem
    calories={300}
    carbs={3}
    fat={15}
    id="2"
    key="2"
    menu_item="chilli - small"
    protein={30}
    restaurant_name="Wendys"
  />
  <MenuItemListItem
    calories={600}
    carbs={80}
    fat={85}
    id="3"
    key="3"
    menu_item="chilli - large"
    protein={50}
    restaurant_name="Burger King"
  />
</div>
`;
